### PR TITLE
Mark items from workqueue as Done()

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Item processor queue used to process catalog items and outputs coalescing for arrays has a bug where it doesn't mark items as done when it finishes processing. This leads to the items being processed again and again wasting resources, clogging the queue and flooding external services with unnecessary requests.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/587

### CODE REVIEW

File: go/tasks/pluginmachinery/workqueue/queue.go [hunk 1]
```diff
@@ -220,32 +220,36 @@ func (q *queue) Start(ctx context.Context) error {
 								logger.Errorf(ctx, "Worker panic'd while processing item [%v]. Error: %v", wrapper.id, e)
 								err = e
 							}
+
+							// Mark the item as done processing. The logic below will determine whether it should go back into the queue
+							// (behind everything else) or whether it requires no further processing.
+							q.queue.Done(item)
```
Comment (kumare3) [inline go/tasks/pluginmachinery/workqueue/queue.go:226]: uh ho! This will case a lot of chaos?

**Comment by kumare3** [comment]
Lets get some unit tests going for this.
